### PR TITLE
poppler: fixes High Sierra build

### PR DESCRIPTION
**Pull request by lukaso**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by lukaso** [comment]
@cho-m can we get this one merged? All looks good. Thanks!

**Review by SMillerDev — CHANGES_REQUESTED** [review]
Patch needs to be submitted upstream

**Comment by lukaso** [comment]
Patch has been submitted upstream, but rejected by maintainer as not something he's willing to support (too old OS version).

**Comment by lukaso** [comment]
This is the diff that would be the patch: https://gitlab.freedesktop.org/lukaso/poppler/-/merge_requests/1

**Comment by lukaso** [comment]
Patch has been removed so this will allow building from High Sierra (10.13). Sierra (10.12) would require the patch (see above).

File: Formula/poppler.rb [hunk 1]
```diff
@@ -58,6 +59,9 @@ class Poppler < Formula
   def install
     ENV.cxx11
 
+    # removes /usr/include from CFLAGS (not clear why)
+    ENV["PKG_CONFIG_SYSTEM_INCLUDE_PATH"] = "/usr/include" if MacOS.version < :mojave
```
Comment (SMillerDev) [inline Formula/poppler.rb:63]: @Rylan12 is this still the way to do this?

Comment (Rylan12) [reply Formula/poppler.rb:63]: Yes, inside the `install` method the `if` statements are the way to go.

`brew style` has also been updated, now, so barring any weird bugs I think if the style checks pass the formula is in good shape (with respect to `on_system` vs `if` statements at least).

File: Formula/poppler.rb [hunk 2]
```diff
@@ -22,6 +22,7 @@ class Poppler < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "glib-utils" => :build
```
Comment (carlocab) [inline Formula/poppler.rb:25]: No longer needed.

Comment (lukaso) [reply Formula/poppler.rb]: Done.

**Review by SMillerDev — APPROVED** [review]

File: Formula/poppler.rb [hunk 3]
```diff
@@ -58,6 +58,9 @@ class Poppler < Formula
   def install
     ENV.cxx11
 
+    # removes /usr/include from CFLAGS (not clear why)
+    ENV["PKG_CONFIG_SYSTEM_INCLUDE_PATH"] = "/usr/include" if MacOS.version < :mojave
```
Comment (carlocab) [inline Formula/poppler.rb:62]: Maybe it wants to try to check the SDK? No idea why, though, as there won't be any `.pc` files to read there.

**Pull request merged**